### PR TITLE
Releasing: scratch dir, performance, ....

### PR DESCRIPTION
**Pull request by pconesa**
Use of scratch dir
Performance:
 Project list load faster
 Bibtex parsing deferred
 Protocols tree loading defer until gui needs it (moved to viewprotocols)
 Delete project accelerated
PyworflowException introduced and capture showing URL for help
Project names do not allow spaces (replaced by "_") to avoid future problems

### CODE REVIEW

**Comment by azazellochg** [comment]
Great start @pconesa , I'll check the PR now.

**Comment by pconesa** [comment]
Great! I didn't mean to bother you, but I'm very happy if you can have a look 2 more eyes and 1 more brain is more than welcome

**Review by fonsecareyna82 — APPROVED** [review]

File: pyworkflow/plugin.py [hunk 1]
```diff
@@ -81,10 +81,13 @@ def registerPlugin(cls, name):
         m._bibtex = {}
         bib = cls.__getSubmodule(name, 'bibtex')
         if bib is not None:
-            try:
-                m._bibtex = pwutils.parseBibTex(bib.__doc__)
-            except Exception:
-                pass
+            if hasattr(bib, "_bibtex"):
+                print("WARNING FOR DEVELOPERS:  %s/%s._bibtex unnecessarily declared. Just the doc string is enough." % (name, "bibtex"))
```
[the remark this answers is not in the document]
Comment (pconesa) [reply pyworkflow/plugin.py:85]: Phenix has a fix in a PR, if all developers behave....it should not be seen.

[the remark this answers is not in the document]
Comment (azazellochg) [reply pyworkflow/plugin.py:85]: Sure, I mean "if hasattr(bib, "_bibtex"):" check should be removed essentially to not complicate the code.

[the remark this answers is not in the document]
Comment (pconesa) [reply pyworkflow/plugin.py:85]: I wish we could have a way to want only developers. Which I think is doable, by "assessing" if the code is a static pure package or a  "pip -e" one.

[the remark this answers is not in the document]
Comment (pconesa) [reply pyworkflow/plugin.py:85]: I see, well, once all is clean and "old" style is gone, is probably safe to remove it.

**Pull request merged**